src/Microsoft.Management.Deployment/Helpers.cpp

```diff
@@ -0,0 +1,92 @@
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+#include "pch.h"
+#include <wil/resource.h>
+#include <wil/win32_helpers.h>
+#include <winrt/Windows.Security.Authorization.AppCapabilityAccess.h>
+#include <appmodel.h>
+
+namespace winrt::Microsoft::Management::Deployment::implementation
+{
+    std::optional<DWORD> GetCallerProcessId()
+    {
+        RPC_STATUS rpcStatus = RPC_S_OK;
+        RPC_CALL_ATTRIBUTES callAttributes = {};
+        callAttributes.Version = RPC_CALL_ATTRIBUTES_VERSION;
+        callAttributes.Flags = RPC_QUERY_CLIENT_PID;
+        rpcStatus = RpcServerInqCallAttributes(nullptr, &callAttributes);
+
+        if ((rpcStatus != RPC_S_NO_CALL_ACTIVE) &&
+            !((rpcStatus == RPC_S_OK) && HandleToULong(callAttributes.ClientPID) == GetCurrentProcessId()))
+        {
+            DWORD callerProcessId = HandleToULong(callAttributes.ClientPID);
+            return callerProcessId;
+        }
+        return {};
+    }
+
+    HRESULT EnsureProcessHasCapability(DWORD callerProcessId)
+    {
+        // Get the caller process id and use it to check if the caller has permissions to access the feature.
+        winrt::Windows::Security::Authorization::AppCapabilityAccess::AppCapabilityAccessStatus status = winrt::Windows::Security::Authorization::AppCapabilityAccess::AppCapabilityAccessStatus::DeniedBySystem;
+
+        auto capability = winrt::Windows::Security::Authorization::AppCapabilityAccess::AppCapability::CreateWithProcessIdForUser(nullptr, L"packageManagement", callerProcessId);
+        status = capability.CheckAccess();
+        RETURN_HR_IF(E_ACCESSDENIED, status != winrt::Windows::Security::Authorization::AppCapabilityAccess::AppCapabilityAccessStatus::Allowed);

```

Comment:
This check works if the caller is packaged but Medium IL? (AKA Centennial)

Reply:
Yes, tested using a hardcoded process id with a centennial app from the store.